src/firecracker/src/main.rs

```diff
@@ -69,14 +70,16 @@ pub fn enable_ssbd_mitigation() {
     }
 }
 
-fn main_exitable() -> FcExitCode {
+fn main_exitable() -> Result<(), FcExitError> {
```

Comment:
`main_exitable` should have a specific error type like `Result<(),MainExitableError>` where:

```rust
#[derive(Debug,thiserror::Error)]
enum MainExitableError {
    // A variant for each possible error that can occur
}
```
this can then be handled where `main_exitable` is called ([which is here](https://github.com/firecracker-microvm/firecracker/blob/main/src/firecracker/src/main.rs#L422))